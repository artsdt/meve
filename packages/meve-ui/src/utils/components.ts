import Vue, { ComponentOptions } from 'vue'
import { CombinedVueInstance } from 'vue/types/vue'

export interface MountComponentApi {
  instance: CombinedVueInstance<any, any, any, any, any>;
  unmount(): void;
}

export function mountComponent(component: ComponentOptions<Vue>, container: string, options = {}): MountComponentApi {
  const instance = new (Vue.extend(component))(options)
  const el = instance.$mount().$el
  const wrapper = document.querySelector(container)!
  wrapper.appendChild(el)

  return {
    instance,
    unmount() {
      instance.$destroy()
      wrapper.removeChild(el)
    },
  }
}

export function nextTick(): Promise<void> {
  return new Promise((resolve) => Vue.nextTick(resolve))
}
