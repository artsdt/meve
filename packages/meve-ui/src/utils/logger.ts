export function warn(message: string, from?: string) {
  if (process?.env?.NODE_ENV === 'development') {
    console.warn(`Warning: ${message} ${from ? `from ${from}` : ''}`)
  }
}
