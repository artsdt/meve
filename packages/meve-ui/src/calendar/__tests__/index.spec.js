import Calendar from '..'
import dayjs from 'dayjs/esm'
import { mount } from '@vue/test-utils'
import { delay } from '../../utils/jest'

test('test calendar v-model', async () => {
  const wrapper = mount(Calendar, {
    propsData: {
      value: '2022-03-17',
    },
  })

  wrapper.vm.current = dayjs('2022-03-17')
  wrapper.vm.now = dayjs('2022-03-17')
  await delay(16)

  await wrapper.find('.m-calendar__date-cell').trigger('click')

  expect(wrapper.emitted().input.length).toBe(1)
  expect(wrapper.emitted().change.length).toBe(1)
  expect(wrapper.emitted().click.length).toBe(1)

  expect(wrapper.emitted().input[0][0]).toBe('2022-02-27')
  expect(wrapper.emitted().change[0][0]).toBe('2022-02-27')
  expect(wrapper.emitted().click[0][0]).toBe('2022-02-27')

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test calendar v-model in range mode', async () => {
  const wrapper = mount(Calendar, {
    propsData: {
      value: [],
      range: true,
    },
  })

  wrapper.vm.current = dayjs('2022-03-17')
  wrapper.vm.now = dayjs('2022-03-17')
  await delay(16)

  await wrapper.find('.m-calendar__date-cell').trigger('click')

  expect(wrapper.emitted().input.length).toBe(1)
  expect(wrapper.emitted().change.length).toBe(1)
  expect(wrapper.emitted().click.length).toBe(1)
  expect(wrapper.emitted().input[0][0]).toStrictEqual(['2022-02-27'])
  expect(wrapper.emitted().change[0][0]).toStrictEqual(['2022-02-27'])
  expect(wrapper.emitted().click[0][0]).toStrictEqual('2022-02-27')

  await wrapper.setProps({ value: ['2022-02-27'] })

  await wrapper.findAll('.m-calendar__date-cell').at(2).trigger('click')

  expect(wrapper.emitted().input.length).toBe(2)
  expect(wrapper.emitted().change.length).toBe(2)
  expect(wrapper.emitted().click.length).toBe(2)
  expect(wrapper.emitted().input[1][0]).toStrictEqual(['2022-02-27', '2022-03-01'])
  expect(wrapper.emitted().change[1][0]).toStrictEqual(['2022-02-27', '2022-03-01'])
  expect(wrapper.emitted().click[1][0]).toStrictEqual('2022-03-01')

  await wrapper.setProps({ value: ['2022-02-27', '2022-03-01'] })
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test calendar exposed methods', async () => {
  const wrapper = mount(Calendar, {
    propsData: {
      value: '2022-03-17',
    },
  })

  wrapper.vm.current = dayjs('2022-03-17')
  wrapper.vm.now = dayjs('2022-03-17')
  await delay(16)

  wrapper.vm.prev()
  expect(wrapper.vm.current.format('YYYY-MM-DD')).toBe('2022-02-17')

  wrapper.vm.next()
  expect(wrapper.vm.current.format('YYYY-MM-DD')).toBe('2022-03-17')

  wrapper.vm.toNow()
  expect(wrapper.vm.current.isSame(dayjs(), 'date')).toBeTruthy()

  wrapper.destroy()
})

test('test calendar header slot', async () => {
  const wrapper = mount(Calendar, {
    propsData: {
      value: '2022-03-17',
    },
    scopedSlots: {
      header: '<div>header slot</div>',
    },
  })

  wrapper.vm.current = dayjs('2022-03-17')
  wrapper.vm.now = dayjs('2022-03-17')
  await delay(16)

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test calendar extra slot', async () => {
  const wrapper = mount(Calendar, {
    propsData: {
      value: '2022-03-17',
    },
    scopedSlots: {
      extra: ({ date }) => {
        return date.date() % 2 === 0 ? 'hello' : 'world'
      },
    },
  })

  wrapper.vm.current = dayjs('2022-03-17')
  wrapper.vm.now = dayjs('2022-03-17')
  await delay(16)

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test calendar disabled', async () => {
  const wrapper = mount(Calendar, {
    propsData: {
      value: '2022-03-17',
      disabled: true,
    },
  })

  wrapper.vm.current = dayjs('2022-03-17')
  wrapper.vm.now = dayjs('2022-03-17')
  await delay(16)

  await wrapper.find('.m-calendar__date-cell').trigger('click')

  expect(wrapper.emitted().input).toBeFalsy()
  expect(wrapper.emitted().change).toBeFalsy()
  expect(wrapper.emitted().click).toBeFalsy()

  wrapper.destroy()
})

test('test calendar readonly', async () => {
  const wrapper = mount(Calendar, {
    propsData: {
      value: '2022-03-17',
      readonly: true,
    },
  })

  wrapper.vm.current = dayjs('2022-03-17')
  wrapper.vm.now = dayjs('2022-03-17')
  await delay(16)

  await wrapper.find('.m-calendar__date-cell').trigger('click')

  expect(wrapper.emitted().input).toBeFalsy()
  expect(wrapper.emitted().change).toBeFalsy()
  expect(wrapper.emitted().click).toBeTruthy()

  wrapper.destroy()
})

test('test calendar custom disabled', async () => {
  const wrapper = mount(Calendar, {
    propsData: {
      value: '2022-03-17',
      customDisabled(date) {
        return date.date() === 27
      },
    },
  })

  wrapper.vm.current = dayjs('2022-03-17')
  wrapper.vm.now = dayjs('2022-03-17')
  await delay(16)

  await wrapper.find('.m-calendar__date-cell').trigger('click')

  expect(wrapper.emitted().input).toBeFalsy()
  expect(wrapper.emitted().change).toBeFalsy()
  expect(wrapper.emitted().click).toBeFalsy()

  await wrapper.findAll('.m-calendar__date-cell').at(1).trigger('click')

  expect(wrapper.emitted().input).toBeTruthy()
  expect(wrapper.emitted().change).toBeTruthy()
  expect(wrapper.emitted().click).toBeTruthy()

  wrapper.destroy()
})
