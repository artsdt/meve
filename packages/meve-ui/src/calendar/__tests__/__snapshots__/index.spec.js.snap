// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test calendar extra slot 1`] = `
"<div class=\\"m-calendar m--box\\">
  <div class=\\"m-calendar__header\\">
    <div class=\\"m-calendar__current\\">2022-03</div>
    <div class=\\"m-calendar__pager\\"><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--primary m-button--shadow m-button--round\\">
        <div class=\\"m-button__content\\"><i class=\\"m-icon m--box meve-icon--set meve-icon-arrow-left\\"></i></div>
      </button><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--primary m-button--shadow m-calendar__today\\" calender-cover=\\"true\\">
        <div class=\\"m-button__content\\">今天</div>
      </button><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--primary m-button--shadow m-button--round\\">
        <div class=\\"m-button__content\\"><i class=\\"m-icon m--box meve-icon--set meve-icon-arrow-right\\"></i></div>
      </button></div>
  </div>
  <div class=\\"m-calendar__calender\\">
    <div class=\\"m-calendar__day-container\\">
      <div class=\\"m-calendar__day\\">周日</div>
      <div class=\\"m-calendar__day\\">周一</div>
      <div class=\\"m-calendar__day\\">周二</div>
      <div class=\\"m-calendar__day\\">周三</div>
      <div class=\\"m-calendar__day\\">周四</div>
      <div class=\\"m-calendar__day\\">周五</div>
      <div class=\\"m-calendar__day\\">周六</div>
    </div>
    <div class=\\"m-calendar__date-container\\">
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">27</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">28</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">1</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">2</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">3</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">4</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">5</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">6</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">7</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">8</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">9</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">10</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">11</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">12</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">13</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">14</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">15</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">16</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text m-calendar--date-today m-calendar--date-picked\\">17</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">18</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">19</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">20</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">21</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">22</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">23</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">24</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">25</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">26</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">27</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">28</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">29</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">30</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">31</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">1</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">2</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">3</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">4</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">5</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">6</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">7</div>world
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">8</div>hello
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">9</div>world
      </div>
    </div>
  </div>
</div>"
`;

exports[`test calendar header slot 1`] = `
"<div class=\\"m-calendar m--box\\">
  <div>header slot</div>
  <div class=\\"m-calendar__calender\\">
    <div class=\\"m-calendar__day-container\\">
      <div class=\\"m-calendar__day\\">周日</div>
      <div class=\\"m-calendar__day\\">周一</div>
      <div class=\\"m-calendar__day\\">周二</div>
      <div class=\\"m-calendar__day\\">周三</div>
      <div class=\\"m-calendar__day\\">周四</div>
      <div class=\\"m-calendar__day\\">周五</div>
      <div class=\\"m-calendar__day\\">周六</div>
    </div>
    <div class=\\"m-calendar__date-container\\">
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">27</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">28</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">1</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">2</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">3</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">4</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">5</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">6</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">7</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">8</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">9</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">10</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">11</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">12</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">13</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">14</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">15</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">16</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text m-calendar--date-today m-calendar--date-picked\\">17</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">18</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">19</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">20</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">21</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">22</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">23</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">24</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">25</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">26</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">27</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">28</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">29</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">30</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">31</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">1</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">2</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">3</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">4</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">5</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">6</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">7</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">8</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">9</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`test calendar v-model 1`] = `
"<div class=\\"m-calendar m--box\\">
  <div class=\\"m-calendar__header\\">
    <div class=\\"m-calendar__current\\">2022-03</div>
    <div class=\\"m-calendar__pager\\"><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--primary m-button--shadow m-button--round\\">
        <div class=\\"m-button__content\\"><i class=\\"m-icon m--box meve-icon--set meve-icon-arrow-left\\"></i></div>
      </button><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--primary m-button--shadow m-calendar__today\\" calender-cover=\\"true\\">
        <div class=\\"m-button__content\\">今天</div>
      </button><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--primary m-button--shadow m-button--round\\">
        <div class=\\"m-button__content\\"><i class=\\"m-icon m--box meve-icon--set meve-icon-arrow-right\\"></i></div>
      </button></div>
  </div>
  <div class=\\"m-calendar__calender\\">
    <div class=\\"m-calendar__day-container\\">
      <div class=\\"m-calendar__day\\">周日</div>
      <div class=\\"m-calendar__day\\">周一</div>
      <div class=\\"m-calendar__day\\">周二</div>
      <div class=\\"m-calendar__day\\">周三</div>
      <div class=\\"m-calendar__day\\">周四</div>
      <div class=\\"m-calendar__day\\">周五</div>
      <div class=\\"m-calendar__day\\">周六</div>
    </div>
    <div class=\\"m-calendar__date-container\\">
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">27</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">28</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">1</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">2</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">3</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">4</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">5</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">6</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">7</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">8</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">9</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">10</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">11</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">12</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">13</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">14</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">15</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">16</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text m-calendar--date-today m-calendar--date-picked\\">17</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">18</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">19</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">20</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">21</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">22</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">23</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">24</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">25</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">26</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">27</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">28</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">29</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">30</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">31</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">1</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">2</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">3</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">4</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">5</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">6</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">7</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">8</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">9</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`test calendar v-model in range mode 1`] = `
"<div class=\\"m-calendar m--box\\">
  <div class=\\"m-calendar__header\\">
    <div class=\\"m-calendar__current\\">2022-03</div>
    <div class=\\"m-calendar__pager\\"><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--primary m-button--shadow m-button--round\\">
        <div class=\\"m-button__content\\"><i class=\\"m-icon m--box meve-icon--set meve-icon-arrow-left\\"></i></div>
      </button><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--primary m-button--shadow m-calendar__today\\" calender-cover=\\"true\\">
        <div class=\\"m-button__content\\">今天</div>
      </button><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--primary m-button--shadow m-button--round\\">
        <div class=\\"m-button__content\\"><i class=\\"m-icon m--box meve-icon--set meve-icon-arrow-right\\"></i></div>
      </button></div>
  </div>
  <div class=\\"m-calendar__calender\\">
    <div class=\\"m-calendar__day-container\\">
      <div class=\\"m-calendar__day\\">周日</div>
      <div class=\\"m-calendar__day\\">周一</div>
      <div class=\\"m-calendar__day\\">周二</div>
      <div class=\\"m-calendar__day\\">周三</div>
      <div class=\\"m-calendar__day\\">周四</div>
      <div class=\\"m-calendar__day\\">周五</div>
      <div class=\\"m-calendar__day\\">周六</div>
    </div>
    <div class=\\"m-calendar__date-container\\">
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text m-calendar--date-picked\\">27</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text m-calendar--in-range\\">28</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text m-calendar--date-picked\\">1</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">2</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">3</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">4</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">5</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">6</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">7</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">8</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">9</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">10</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">11</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">12</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">13</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">14</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">15</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">16</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text m-calendar--date-today\\">17</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">18</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">19</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">20</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">21</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">22</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">23</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">24</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">25</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">26</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">27</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">28</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">29</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">30</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">31</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">1</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">2</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">3</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">4</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">5</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">6</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">7</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">8</div>
      </div>
      <div class=\\"m-calendar__date-cell\\">
        <div class=\\"m-calendar__date-text\\">9</div>
      </div>
    </div>
  </div>
</div>"
`;
